### PR TITLE
Billing Telemetry

### PR DESCRIPTION
**Pull request by MauAraujo**
## What does this PR do?

This PR adds telemetry spans to all `StripeBillingManager` functions, adds telemetry wrappers for errors and adds attributes relevant to the billing endpoints. After this is merged I will use the telemetry spans to create alerts when customers setup credit cards.

honeycomb_1
honeycomb_2

### CODE REVIEW

File: internal/billing/stripe.go [hunk 1]
```diff
@@ -19,7 +21,10 @@ type StripeBillingManager struct {
...
 
 // CreateCustomer will create a customer in Stripe only if the project doesn't have a BillingID
-func (s *StripeBillingManager) CreateCustomer(userEmail string, proj *models.Project) (customerID string, err error) {
+func (s *StripeBillingManager) CreateCustomer(ctx context.Context, userEmail string, proj *models.Project) (customerID string, err error) {
+	ctx, span := telemetry.NewSpan(ctx, "create-stripe-customer")
+	defer span.End()
+
 	stripe.Key = s.StripeSecretKey
 
 	if proj.BillingID == "" {
```
Comment (ferozemohideen) [inline internal/billing/stripe.go:30]: should do telemetry.Attribute for billing id here

File: internal/billing/stripe.go [hunk 2]
```diff
@@ -43,22 +48,28 @@ func (s *StripeBillingManager) CreateCustomer(userEmail string, proj *models.Pro
...
 
 // DeleteCustomer will delete the customer from the billing provider
-func (s *StripeBillingManager) DeleteCustomer(proj *models.Project) (err error) {
+func (s *StripeBillingManager) DeleteCustomer(ctx context.Context, proj *models.Project) (err error) {
+	ctx, span := telemetry.NewSpan(ctx, "delete-stripe-customer")
+	defer span.End()
+
 	stripe.Key = s.StripeSecretKey
 
 	if proj.BillingID != "" {
```
Comment (ferozemohideen) [inline internal/billing/stripe.go:57]: log billing id to telemetry

File: internal/billing/stripe.go [hunk 3]
```diff
@@ -148,26 +168,32 @@ func (s *StripeBillingManager) SetDefaultPaymentMethod(paymentMethodID string, p
...
 	_, err = paymentmethod.Detach(paymentMethodID, nil)
 	if err != nil {
-		return err
+		return telemetry.Error(ctx, span, err, "failed to delete Stripe payment method")
 	}
 
 	return nil
 }
 
 // GetPublishableKey is a no-op
```
Comment (ferozemohideen) [inline internal/billing/stripe.go:192]: update this comment

**Review by ferozemohideen — APPROVED** [review]
for all the lint errors you are getting, you can change `ctx, span :=` to `_, span :=` if you are not using the ctx

File: internal/billing/stripe.go (hunk 1 above)
Comment (MauAraujo) [reply internal/billing/stripe.go:34]: Added in the latest commit

File: internal/billing/stripe.go (hunk 2 above)
Comment (MauAraujo) [reply internal/billing/stripe.go:65]: Added in the latest commit

File: internal/billing/stripe.go (hunk 3 above)
Comment (MauAraujo) [reply internal/billing/stripe.go]: Thanks, didn't notice it

**Pull request merged**